mdpow/tests/test_ensemble.py

```diff
@@ -149,6 +149,21 @@ def _conclude_universe(self):
         TestRun = TestAnalysis(Sim).run(start=0, step=1, stop=10)
         assert Sim.keys() == TestRun.key_list
 
+    def test_ensemble_analysis_run(self):
+        class TestAnalysis(EnsembleAnalysis):
+            def __init__(self, test_ensemble):
+                super(TestAnalysis, self).__init__(test_ensemble)
+
+            def test_single_frame(self):
+                with pytest.raises(NotImplementedError) as excinfo:
+                    TestAnalysis._single_frame(self)
+                assert excinfo.type == 'NotImplementedError'
+
+            def test_single_universe(self):
+                with pytest.raises(NotImplementedError) as excinfo:
+                    TestAnalysis._single_universe(self)
+                assert excinfo.type == 'NotImplementedError'
+
```

Comment:
#216 https://github.com/Becksteinlab/MDPOW/pull/216#discussion_r1042832663

@orbeckst Can you provide more detail for the linked comment? This test might not have shown up with the recent commit, but if it did, can you elaborate on how it is not correctly testing the base class?

Reply:
Codecov indicates that these lines were not tested. Codecov is not always right but it's something to look into carefully. Come up with a way to convince yourself (and me) that your tests really test the code that you think they do. Perhaps temporarily change the raise NotImplementError to ValueError and then your test should fail. Report back on what you did.

Reply:
I changed exception/error types and nothing changed. If the test and class definitions were nested correctly, pytest would have also picked up that 
`assert excinfo.type ==` should have been `assert excinfo.type **is**`, so basically nothing was happening. There is a way to match directly to string but I did it incorrectly and this way is simpler.

I split the single test into two separate, one for `_singe_frame` and one for `_single_universe`. Each has its own TestAnalysis class definition, where it shows that the run method not being tested is defined, so the undefined run method is explicitly tested for raising NotImplementedError, for each case. 

There is probably a way to consolidate the two tests back into one, but it is very clear what is being done this way. It still seems there is a simpler method.  I will review again in the morning and make changes requested for documentation so this can be merged if everything is correct otherwise!